File: mask-operations/blending.cpp
//-----------------------------------------------------------------------------
//
// Example for images blending in C++.
//
//-----------------------------------------------------------------------------

#include <opencv2/opencv.hpp>

int main() {
    cv::Mat img1, img2, result;

    img1 = cv::imread("../img.jpg", cv::IMREAD_COLOR);
    img2 = cv::imread("../img.jpg", cv::IMREAD_COLOR);

    // Make sence for blending.
    cv::flip(img1, img2, 0);

    // Blend for 50%.
    cv::addWeighted(img1, 0.5, img2, 0.5, 0.0, result);

    cv::namedWindow("blending");
    cv::imshow("blending", result);

    cv::waitKey();
    cv::destroyAllWindows();
}

File: mask-operations/blending/blending.py
#-----------------------------------------------------------------------------
#
# Example for images blending in Python with OpenCV.
#
#-----------------------------------------------------------------------------

import cv2

img1 = cv2.imread("../img.jpg", cv2.IMREAD_COLOR)
img2 = cv2.imread("../img.jpg", cv2.IMREAD_COLOR)

img2 = cv2.flip(img1, 0)

# Blend for 50%.
result = cv2.addWeighted(img1, 0.5, img2, 0.5, 0.0)

cv2.namedWindow("blending")
cv2.imshow("blending", result)

cv2.waitKey(0)
cv2.destroyAllWindows()
